magpie/alembic/versions/2023-11-10_91af68bcdc67_case_insensitive_user_name_constraint.py

```diff
@@ -0,0 +1,44 @@
+"""
+Case Insensitive User_Name Constraint
+
+Revision ID: 91af68bcdc67
+Revises: 5e5acc33adce
+Create Date: 2023-11-10 15:58:23.068465
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.orm.session import sessionmaker
+
+# Revision identifiers, used by Alembic.
+# pylint: disable=C0103,invalid-name  # revision control variables not uppercase
+revision = "91af68bcdc67"
+down_revision = "5e5acc33adce"
+branch_labels = None
+depends_on = None
+
+Session = sessionmaker()
+
+users = sa.table(
+    "users",
+    sa.column("id", sa.Integer),
+    sa.column("user_name", sa.String),
+)
+
+
+def upgrade():
+    op.create_index(
+        "ix_users_user_name_unique_case_insensitive",
+        "users",
+        [sa.text("lower(user_name)")],
+        unique=True,
+    )
+    session = Session(bind=op.get_bind())
+    for user in session.execute(sa.select(users)):
+        session.execute(users.update().where(users.c.id == user.id).values(user_name=user.user_name.lower()))
```

Comment:
I think this could be done with a single query: 
```python
session.execute(users.update().values({users.c.id: func.lower(users.c.id)})
```

Reply:
I'm not sure I understand this query... 

Did you mean: 

```python
session.execute(users.update().values({users.c.user_name: func.lower(users.c.user_name)}))
```

That seems to work. I'll update it